migrations/20230410210224-add-UserTwoFactorMethod-table.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable('UserTwoFactorMethods', {
+      id: {
+        type: Sequelize.INTEGER,
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+      },
+      method: {
+        type: Sequelize.STRING,
+        allowNull: false,
+      },
+      data: {
+        type: Sequelize.JSONB,
+        allowNull: true,
+      },
+      UserId: {
+        type: Sequelize.INTEGER,
+        references: { key: 'id', model: 'Users' },
+        allowNull: true,
+        onDelete: 'CASCADE',
+        onUpdate: 'CASCADE',
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.fn('now'),
+      },
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
+        defaultValue: Sequelize.fn('now'),
+      },
+      deletedAt: {
+        allowNull: true,
+        type: Sequelize.DATE,
+      },
+    });
+
+    await queryInterface.addIndex('UserTwoFactorMethods', ['UserId', 'method']);
```

Comment:
Since `userMethods` query by `UserId` only, perhaps we should also index for that:
```suggestion
    await queryInterface.addIndex('UserTwoFactorMethods', ['UserId', 'method']);
     await queryInterface.addIndex('UserTwoFactorMethods', ['UserId']);
```

Reply:
Well, considering that we shouldn't have many methods per user, I think it would even be OK to only index by `UserId` here.